File: DSRFQ.Web/Modules/ServerTypes/Wallet/WalletTransactionItemsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WalletTransactionItemsRow } from "./WalletTransactionItemsRow";

export interface WalletTransactionItemsColumns {
    Id: Column<WalletTransactionItemsRow>;
    WalletTransactionStripeSessionId: Column<WalletTransactionItemsRow>;
    ItemId: Column<WalletTransactionItemsRow>;
    Description: Column<WalletTransactionItemsRow>;
    ProductId: Column<WalletTransactionItemsRow>;
    PriceId: Column<WalletTransactionItemsRow>;
    AmountDiscount: Column<WalletTransactionItemsRow>;
    AmountSubtotal: Column<WalletTransactionItemsRow>;
    AmountTax: Column<WalletTransactionItemsRow>;
    AmountTotal: Column<WalletTransactionItemsRow>;
    Quantity: Column<WalletTransactionItemsRow>;
    Currency: Column<WalletTransactionItemsRow>;
    InsertDate: Column<WalletTransactionItemsRow>;
    InsertBy: Column<WalletTransactionItemsRow>;
    UpdateDate: Column<WalletTransactionItemsRow>;
    UpdateBy: Column<WalletTransactionItemsRow>;
    DeleteDate: Column<WalletTransactionItemsRow>;
    DeleteBy: Column<WalletTransactionItemsRow>;
}

export class WalletTransactionItemsColumns extends ColumnsBase<WalletTransactionItemsRow> {
    static readonly columnsKey = 'Wallet.WalletTransactionItems';
    static readonly Fields = fieldsProxy<WalletTransactionItemsColumns>();
}

File: DSRFQ.Web/Modules/Wallet/WalletTransactionItems/WalletTransactionItemsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Wallet.Columns;

[ColumnsScript("Wallet.WalletTransactionItems")]
[BasedOnRow(typeof(WalletTransactionItemsRow), CheckNames = true)]
public class WalletTransactionItemsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string WalletTransactionStripeSessionId { get; set; }
    [EditLink]
    public string ItemId { get; set; }
    public string Description { get; set; }
    public string ProductId { get; set; }
    public string PriceId { get; set; }
    public decimal AmountDiscount { get; set; }
    public decimal AmountSubtotal { get; set; }
    public decimal AmountTax { get; set; }
    public decimal AmountTotal { get; set; }
    public decimal Quantity { get; set; }
    public string Currency { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}